packages/block-editor/src/components/navigable-toolbar/index.js

```diff
@@ -65,6 +65,12 @@ class NavigableToolbar extends Component {
 		if ( this.props.focusOnMount ) {
 			this.focusToolbar();
 		}
+
+		this.toolbar.current.addEventListener( 'keydown', this.switchOnKeyDown );
```

Comment:
I guess these are temporary solutions to allow us to move forward with the proposed changes? Is it strictly incompatible that this could be kept as using `onKeyDown` via React binding?

Reply:
That is not really a temporary solution. For me, this component should have been written this way from the start. (So does WritingFlow but I didn't change writing flow in this PR because of impact on RichText, see previous back and forth).

The idea is that these components do care about their DOM children and not their React children. A navigable container is a way to enable arrow/tab navigation between their children in terms of DOM and don't really care about how that DOM element got rendered (which specific React tree renders it).

Reply:
Could more comments be added inline about why we cannot use React events?